corehq/apps/hqwebapp/templates/hqwebapp/includes/new_app_version_modal.html

```diff
@@ -0,0 +1,21 @@
+{% load i18n %}
+
+<div class="modal fade" id ="newAppVersionModal" tabindex="-1" role="dialog">
+    <div class="modal-dialog" role="document">
+      <div class="modal-content">
+        <div class="modal-header">
+          <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+          <h4 class="modal-title">{% trans "A new version of this app is available" %}</h4>
+        </div>
+        <div class="modal-body">
+          <p>{% trans "Please click on \"Get Latest App\" to get on the newer version. It's recomended that you work on the latest version of the app." %}</p>
+          <p> {% trans "Please note that all unsaved changes would be lost while moving to new version. You can submit the changes and manually refresh this page which will automatically take you to latest version."%}</p>
```

Comment:
I wonder if this message could be a bit more explicit. If the user is in a form, they'll want to either submit it, or refresh and then find it under incomplete forms. If they're not in a form, I don't think they'll lose anything meaningful. Maybe the product team would have input?

Reply:
Thanks for your inputs on it. I myself was not that confident on the text but wanted the users to know that it may result in the loss of their currently unsaved work. I think it would be a good idea to ping product.

> It'd be cool if you detected whether or not they're in a form and instructed them accordingly. I think you'd need to look for particular content on the page, the submit button or something, to detect that.

I have very little context on WebApps, do you have some suggestions on how this can be achieved? This would  surely be a really cool thing to have.

Reply:
I was thinking of looking at the page's HTML - one possibility would be to check if a div with the class `webforms-nav-container` is visible on the page, as that's the class for the main container in the [web apps form entry template](https://github.com/dimagi/commcare-hq/blob/7df1abfec6171c89d0fb663d8d515cddf651e95d/corehq/apps/cloudcare/templates/form_entry/templates.html#L104-L105).

Reply:
Tried this https://github.com/dimagi/commcare-hq/pull/30778/commits/e674a719ed266e6cb5ad2c9d1b73ee1be5f0a4b8

Reply:
I missed this earlier, new commit looks good.

One question on the message: the text says "manually refresh this page which will automatically take you to latest version" but if they refresh the page and there's a new app build, won't they get a permission denied error? I thought they'd have to go back to the web apps home screen, by clicking "Web Apps" in the top left.

Reply:
Ohh! This is a good point, Jenny. I think we should update it to - 

> You can submit the changes and click on "Web Apps" in the top left of the page which will list the latest version of this app.

I will update the Product about the change in the wording.